### PR TITLE
Enable Cypress test selection in the `master` branch in CI

### PR DESCRIPTION
**Pull request by ddzz**
## Description
This PR enables test selection in `master` branch CI. Files changed between the current commit and previous commit are used in test selection.

## Original issue(s)
https://github.com/department-of-veterans-affairs/va.gov-team/issues/37921

## Acceptance criteria
- [ ] CI passes.

### CODE REVIEW

**Comment by timwright12** [comment]
deferring to @theodur for this review

File: .github/workflows/continuous-integration.yml
Comment (holdenhinkle) [inline .github/workflows/continuous-integration.yml:884]: I don't have time to do a full review right now, but a bunch of the graphs in DOMO are generated from queries against `IS_MASTER_BUILD` being `true` or `false`:

image

These graphs will definitely be affected by `IS_MASTER_BUILD` always being `false`. Which is fine--they can be changed. Maybe we can talk about this tomorrow.

Thanks.

File: .github/workflows/continuous-integration.yml
Comment (theodur) [inline .github/workflows/continuous-integration.yml:54]: Is it possible to combine this into a composite action with the other `List changed files` step, and have the action accept an `is_master` input? That way, we'd only need one `changed-files` step. Or maybe we can have one `changed-files` step and handle the logic in bash rather than GHA.

Having a single step for getting the changed files would also prevent us from needing two `get-changed-apps` steps and two `Build` steps

File: .github/workflows/continuous-integration.yml
Comment (theodur) [inline .github/workflows/continuous-integration.yml:82]: This would turn changed app builds on in `master`, which we don't want to do yet. If we're able to get a single `changed-files` step like I mentioned above, we can remove this and the duplicate `get-changed-apps` step, and get the `app_entries` from that single `get-changed-apps` step

File: .github/workflows/continuous-integration.yml
Comment (ddzz) [reply .github/workflows/continuous-integration.yml:884]: Since we won't be running the full test suite in CI, `IS_MASTER_BUILD` needs to be `false` here. The full suite will run in the new `e2e-tests.yml` workflow, so I've set `IS_MASTER_BUILD` to `true` in that file: https://github.com/department-of-veterans-affairs/vets-website/pull/20413/files#diff-194218c48b9a0cdd03974145733804c2d992ca818529fe2fa69a501d8b5b1cc3R355

File: .github/workflows/e2e-tests.yml
Comment (ddzz) [reply .github/workflows/e2e-tests.yml:146]: This line can be removed. Since test selection is disabled in this workflow, `needs.cypress-tests-prep.outputs.tests` will never be an empty array.

**Comment by ddzz** [comment]
@theodur Is the new E2E workflow working as expected, or is it still being tested? I think we shouldn't merge this PR until that's ready.

File: .github/workflows/e2e-tests.yml
Comment (theodur) [reply .github/workflows/e2e-tests.yml:146]: Cool, that makes sense ✅

**Comment by theodur** [comment]
> @theodur Is the new E2E workflow working as expected, or is it still being tested? I think we shouldn't merge this PR until that's ready.

@ddzz The E2E workflow is complete, and should be working as expected

File: .github/workflows/continuous-integration.yml
Comment (theodur) [inline .github/workflows/continuous-integration.yml:386]: The same thing I mentioned in this [comment](https://github.com/department-of-veterans-affairs/vets-website/pull/20413#r819194274) applies here also. If we could combine the two `List changed files` steps into one, it would prevent the need for an extra `Set TESTS variable` step

File: .github/workflows/continuous-integration.yml
Comment (theodur) [inline .github/workflows/continuous-integration.yml:385]: I noticed that with `git diff --name-only HEAD~`, if a file is renamed/moved in a commit, it only shows that the new file path was changed, but not the old one.

e.g. when moving `src/platform/site-wide/banners/index.js` -> `src/platform/new-path/banners/index.js` in a commit:

`git diff --name-only HEAD~` only shows that `src/platform/new-path/banners/index.js` was changed, and not `src/platform/site-wide/banners/index.js`. 

This probably won't be an issue 99% of the time, but if that change was renaming/moving 
`src/platform/site-wide/banners/index.js` -> `src/applications/static-pages/banners/index.js` for example, platform tests wouldn't run, even though a file in `src/platform` was changed. Granted, linting would catch if something was wrong on the import side, it's unknown how that would affect functionality if untested.

I use `git diff-tree --no-commit-id --name-only -r ${{ commit_sha }}` in the [E2E workflow](https://github.com/department-of-veterans-affairs/vets-website/blob/master/.github/workflows/e2e-tests.yml#L57) for getting the changed files on `master`. Using `git diff-tree` seems to include both the source and destination renamed files. Not sure if there's a way to get that functionality with `git diff`.

Comment (ddzz) [reply .github/workflows/continuous-integration.yml]: Thanks for reviewing. I don't think it's possible to do this with `git diff` so I updated the PR with `diff-tree`.

File: .github/workflows/continuous-integration.yml
Comment (theodur) [inline .github/workflows/continuous-integration.yml:385]:
```suggestion
            echo ::set-output name=all_changed_files::$(git diff-tree --no-commit-id --name-only -r ${{ github.sha }})
```
Since `commit_sha` isn't defined here

File: .github/workflows/e2e-tests.yml
Comment (theodur) [inline .github/workflows/e2e-tests.yml:355]:
```suggestion
          RUN_FULL_SUITE: true
```

Comment (theodur) [reply .github/workflows/e2e-tests.yml:355]: Whoops, disregard this one, didn't realize this was in the `testing-reports-cypress` job

**Review by theodur — APPROVED** [review]
These changes look good, thanks for getting this done so quickly! I'll defer to @holdenhinkle for the test selection specific changes, although it doesn't look like much changed in that script

**Review by holdenhinkle — APPROVED** [review]

**Pull request merged**
